File: src/gncgym/envs/supplyShipModel/guidance.py
import numpy as np
from math import atan2


def straightLineGuidance(state, surge_des=4, lookahead=75):
    # Unpack the state
    x, y, _, _, _, _ = state

    # Standard Line-of-sight for straight line path following
    heading_des = atan2(-y, lookahead)

    return np.array([surge_des, heading_des])

File: src/gncgym/envs/supplyShipModel/matlab/guidance.m
function [ state_out ] = guidance( state_in )
%Desired u and ps
global Delta u_des

y = state_in(2);
%Standard LOS for straight line path following
psi_des = atan2(-y,Delta);

%Return correct state
state_out = [u_des;psi_des];

end

